src/Microsoft.AspNetCore.Razor.Tools/GenerateCommand.cs

```diff
@@ -166,9 +166,12 @@ protected override bool ValidateArguments()
                 if (result.CSharpDocument.Diagnostics.Count > 0)
                 {
                     success = false;
-                    foreach (var error in result.CSharpDocument.Diagnostics)
+
+                    // Only show the first 100 errors to prevent massive string allocations.
+                    for (var i = 0; i < result.CSharpDocument.Diagnostics.Count && i < 100; i++)
                     {
-                        Console.Error.WriteLine(error.ToString());
+                        var error = result.CSharpDocument.Diagnostics[i];
+                        Error.WriteLine(error.ToString());
                     }
```

Comment:
Can you add an additional error message if there are over 100 errors? Its important that this experience be polished.